src/data_structures.rs

```diff
@@ -77,29 +64,89 @@ impl<
 /* ************************************************************************* */
 /* ************************************************************************* */
 
-/// Proving key for a specific index (i.e., R1CS matrices).
-pub struct IndexProverKey<
-    'a,
+/// Verification key, prepared (preprocessed) for use in pairings.
+pub struct PreparedIndexVerifierKey<F: PrimeField, PC: PolynomialCommitment<F, DensePolynomial<F>>>
```

Comment:
Review needed: prepared index verifier key

Reply:
This looks good!